debugger/io.c

```diff
@@ -0,0 +1,74 @@
+/*
+  MSP430 Emulator
+  Copyright (C) 2020 Rudolf Geosits (rgeosits@live.esu.edu)
+
+  "MSP430 Emulator" is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  "MSP430 Emulator" is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "io.h"
+#include "websockets/packet_queue.h"
+
+void print_serial (Emulator *emu, char *buf)
+{
+    switch (emu->mode)
+    {
+        case Emulator_Mode_Web:
+            packet_enqueue(emu, buf, strlen(buf) + 1, SERIAL_PACKET_OPCODE);
+            break;
+        case Emulator_Mode_Cli:
+            // This is just for fallback or compatibility, if another UART like peripheral
+            // was implemented independently
+            printf("Serial %s\n", buf);
+            break;
+    }
+}
+
+void put_serial(Emulator* const emu, const uint8_t x)
+{
+    if (emu->usci_output_pipe == NULL)
+    {
+        char str[2];
+        str[0] = (char)x;
+        str[1] = 0;
+        print_serial(emu, str);
+    }
+    else
+    {
+        const ssize_t count = write(emu->usci_output_pipe_fd, &x, sizeof(uint8_t));
+        fflush(emu->usci_output_pipe);
+        if (count != 1)
+            print_console(emu, "Cannot write to serial output\n");
```

Comment:
It's a good practise to add braces { } around blocks.

Reply:
strong disagree - for single line `{}` are not necessary :)

Reply:
https://github.com/lefticus/cppbestpractices/blob/master/03-Style.md#-are-required-for-blocks

Reply:
hmm, first time I see this repo, so I do not consider it as reference - it's just someones opinion. And there are other opinions:

https://www.kernel.org/doc/html/v5.6/process/coding-style.html#placing-braces-and-spaces
Linux Kernel, you might've heard of it :D

https://google.github.io/styleguide/cppguide.html#Conditionals
Google, also something you might know

:P